v1_20/src/main/java/com/denizenscript/denizen/nms/v1_20/impl/network/handlers/DenizenNetworkManagerImpl.java

```diff
@@ -309,1036 +281,75 @@ public void send(Packet<?> packet, PacketSendListener genericfuturelistener) {
         }
         packetsSent++;
         if (packet instanceof ClientboundBundlePacket bundlePacket) {
-            Iterator<Packet<ClientGamePacketListener>> iter = bundlePacket.subPackets().iterator();
-            int count = 0;
-            while (iter.hasNext()) {
-                count++;
-                if (processPacket(iter.next(), genericfuturelistener)) {
-                    iter.remove();
-                    count--;
+            List<Packet<ClientGamePacketListener>> processedPackets = new ArrayList<>();
+            for (Packet<ClientGamePacketListener> subPacket : bundlePacket.subPackets()) {
+                Packet<ClientGamePacketListener> processed = processPacketHandlersFor(subPacket);
+                if (processed != null) {
+                    processedPackets.add(processed);
                 }
             }
-            if (count == 0) {
+            if (processedPackets.isEmpty()) {
                 return;
             }
+            packet = new ClientboundBundlePacket(processedPackets);

```

Comment:
This will *always* alter the packet - it should *only* alter the packet if one of the called methods needs an alteration.

Reply:
Added basic change tracking to only change the bundle packet when any of the sub-packets were changed, let me know if that looks good